src/actions/error-action-creators.js

```diff
@@ -0,0 +1,80 @@
+import get from 'lodash/get'
+import merge from 'lodash/merge'
+import statusCode from '../constants/status-code'
+
+const _ = {
+  get,
+  merge
+}
+
+/**
+ *  actionError type definition
+ *  @typedef {Object} ErrorAction
+ *  @property {string} type - Action error type
+ *  @property {string} payload
+ *  @property {string} payload.statusCode - Response status code
+ *  @property {string} payload.message - Error message
+ *  @property {object} payload.error - Error object
+*/
+
+/**
+ * @param {Object} [invalidFields] - object with invalid fields and resaons
+ * @param {string} failActionType
+ * @returns {ErrorAction}
+ */
+function handleValidationFailureBeforeRequest(invalidFields, failActionType) {
+  return {
+    type: failActionType,
+    payload: {
+      message: 'data validation failed before the http request was make',
+      data: invalidFields,
+      statusCode: statusCode.badRequest
+    }
+  }
+}
+
+/**
+ *  @param {Object} [err={}] - Error returned from axios
+ *  @param {string} failActionType
+ *  @returns {ErrorAction}
+ */
+function handleAxiosError(err = {}, failActionType) {
+  const failActionSharedProperties = {
+    type: failActionType,
+    payload: {
+      message: err.message
+    }
+  }
+  if (err.response) {
+    if (err.response.status === statusCode.badRequest) {
+      return _.merge({}, failActionSharedProperties, {
+        message: 'response bad request due to data validation failures',
+        data: _.get(err, 'response.data')
+      })
+    }
+    return _.merge({}, failActionSharedProperties, {
+      payload: {
+        data: _.get(err, 'response.data'),
+        message: _.get(err, 'response.data.message'),
+        statusCode: err.response.status
+      }
+    })
+  } else if (err.request) {
+    return _.merge({}, failActionSharedProperties, {
+      payload: {
+        message: 'request was made but no response was received'
+      }
+    })
+  } else {
+    return _.merge({}, failActionSharedProperties, {
+      payload: {
+        message: 'error happened in setting up the request that triggered an error'
```

Comment:
Do we need merge here?
Since` failActionSharedProperties` is a pretty simple object,
could we just write the return object directly, like 
```
return {
  type: failActionType,
  payload: {
    data: _.get(err, 'response.data'),
    message: _.get(err, 'response.data.message'),
    statusCode: err.response.status
  }
}
```

Reply:
I'll replace it to literal declaration.